django_date_extensions/fields.py

```diff
@@ -114,12 +139,28 @@ def to_python(self, value):
         if value == 'past':
             return ApproximateDate(past=True)
 
-        if not ansi_date_re.search(value):
-            raise ValidationError('Enter a valid date in YYYY-MM-DD format.')
+        prefix = None
+
+        if isinstance(value, (datetime.date, datetime.datetime)):
```

Comment:
You'll be pleased to know that this isn't necessary on top of the new version 2, because we're now using `get_prep_value`. The commit at 4db71902014ebe5c6095c746170be67472deab95 includes those tests so you can see that it does still work.